### PR TITLE
[Bugfix Beta] Fix attrs in className and attribute bindings

### PR DESCRIPTION
**Pull request by chadhietala**

### CODE REVIEW

**Comment by chadhietala** [comment]
Fixes #14381

File: packages/ember-glimmer/lib/utils/bindings.js [hunk 1]
```diff
@@ -14,6 +14,17 @@ function referenceForKey(component, key) {
...
 function referenceForParts(component, parts) {
+  let isAttrs = parts[0] === 'attrs';
+
+  // TODO deprecate this
+  if (isAttrs) {
+    parts.shift();
+
+    if (parts.length === 1) {
+      return referenceForKey(component, parts[0]);
+    }
```
Comment (krisselden) [inline packages/ember-glimmer/lib/utils/bindings.js]: what is the else here? is it supposed to fall through or assert?

File: packages/ember-glimmer/lib/utils/bindings.js [hunk 2]
```diff
@@ -14,6 +14,17 @@ function referenceForKey(component, key) {
 }
 
 function referenceForParts(component, parts) {
+  let isAttrs = parts[0] === 'attrs';
```
Comment (krisselden) [inline packages/ember-glimmer/lib/utils/bindings.js]: is parts guaranteed to be length 1? is it very important that we never index outside of the array on get.

Comment (chadhietala) [inline packages/ember-glimmer/lib/utils/bindings.js]: Upstream from this [the check is done](https://github.com/emberjs/ember.js/blob/473a4f5f05423837c70376d8c12a5206aa2f8629/packages/ember-glimmer/lib/utils/bindings.js#L66-L67) to make sure we have a path reference.

File: packages/ember-glimmer/lib/utils/bindings.js (hunk 1 above)
Comment (chadhietala) [inline packages/ember-glimmer/lib/utils/bindings.js]: We want this to fall through. So given `attrs.foo` we want to delegate to `referenceForKey`, however if we shift off the value and we still have a path e.g. `parts.length !== 1` (guranteed to never be 0) then we want to still create a path ref.

**Pull request merged**